core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2204,27 +2203,37 @@ class ReplicaManager(val config: KafkaConfig,
 
           completeDelayedFetchOrProduceRequests(tp)
```

Comment:
Hmm, as I'm looking over the code, this seems wrong. We'd want to trigger delayed operations after the call to `makeFollower` since the change from leader to follower is what would complete the operation.

Reply:
Yes. You are correct. Let me write a test for it.

Reply:
Done. Verified that the new tests fail without the changes to replica manager.